frontend/src/hooks/useDownload.ts

```diff
@@ -0,0 +1,47 @@
+import { useState } from 'react';
+import { photoService } from '../apis/services/photo.service';
+import { DEBUG_MESSAGES } from '../constants/debugMessages';
+import { mockSpaceData } from '../pages/manager/spaceHome/mockSpaceData';
+import { tryAsync } from '../utils/tryAsync';
+
+interface UseDownloadProps {
+  spaceName: string;
+}
+
+const useDownload = ({ spaceName }: UseDownloadProps) => {
+  const [isDownloading, setIsDownloading] = useState(false);
+
+  const downloadBlob = (blob: Blob) => {
+    const url = URL.createObjectURL(blob);
+
+    const a = document.createElement('a');
+    document.body.appendChild(a);
+    a.href = url;
+    a.download = `${spaceName}.zip`;
+    a.click();
+
+    document.body.removeChild(a);
+    window.URL.revokeObjectURL(url);
+  };
+
+  const handleDownload = () => {
+    setIsDownloading(true);
+    tryAsync(
+      async () => {
+        setIsDownloading(true);
+        const response = await photoService.downloadZip(mockSpaceData.code);
+        const blob = response.data;
+        if (!blob) {
+          throw new Error(DEBUG_MESSAGES.NO_BLOB);
+        }
```

Comment:
```suggestion
        if (!(blob instanceof Blob)) {
	  // ...
	}
```

blob이 Blob 인스턴스인지 확인하는 로직을 추가하면 어떨까요? 더 안전하고 정확한 타입을 검증할 수 있을 것 같아요.!